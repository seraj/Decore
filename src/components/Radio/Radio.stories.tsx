import { text, withKnobs, boolean } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";
import * as React from "react";
import wInfo from "../../utils/wInfo";
import Radio from "./Radio";

const stories = storiesOf("Components/Radio", module);
stories.addDecorator(withKnobs);

stories.add(
  "default",
  wInfo()(() => (
    <>
      <div>
        <Radio
          checked={boolean("checked", false)}
          disabled={boolean("disabled", false)}
          name="test"
        />
      </div>
      <div>
        <Radio name="test" />
      </div>
    </>
  ))
);
