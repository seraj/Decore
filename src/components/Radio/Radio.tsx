import * as React from "react";
import StyledRadio, { HiddenRadio, RadioContainer } from "./styled/StyledRadio";
import RadioProps from "./Radio.props";

const Radio: React.FC<RadioProps> = ({
  className,
  checked,
  children,
  disabled,
  ...props
}) => (
  <RadioContainer className={className}>
    <HiddenRadio
      {...props}
      checked={checked}
      disabled={disabled}
      type="radio"
    />
    <StyledRadio checked={checked} disabled={disabled} />
  </RadioContainer>
);

export default Radio;
