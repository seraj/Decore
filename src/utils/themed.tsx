import * as React from "react";
import * as renderer from "react-test-renderer";
import Decore from "../components/Decore";
import theme from "../theme";

const themed = (component: JSX.Element) =>
  renderer.create(<Decore theme={theme}>{component}</Decore>);

export default themed;
