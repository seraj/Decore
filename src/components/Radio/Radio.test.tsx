import * as React from "react";

import themed from "../../utils/themed";
import Radio from "./Radio";

describe("Radio", () => {
  it("should render correctly", () => {
    const component = themed(<Radio />).toJSON();

    expect(component).toMatchSnapshot();
  });
});
