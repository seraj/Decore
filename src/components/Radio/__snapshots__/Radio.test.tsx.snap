// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should render correctly 1`] = `
.c2 {
  border: 1.5px solid #999999;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  background-color: #ffffff;
  padding: 1rem 0.8rem;
  height: 42px;
  border-radius: 3px;
  color: #666666;
}

.c2:hover {
  border-color: #333333;
  color: #666666;
}

.c2:active {
  color: #333333;
}

.c2:focus {
  color: #333333;
  border-color: #ffffff;
}

.c2:disabled {
  border: none;
  background: #E5E5E5;
  color: #999999;
  cursor: not-allowed !important;
  pointer-events: none;
}

.c0 {
  display: inline-block;
  vertical-align: middle;
  position: relative;
  cursor: pointer;
}

.c1 {
  position: absolute;
  opacity: 0;
  cursor: pointer;
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  height: 20px;
  width: 20px;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  background-color: #ffffff;
  border-radius: 10px;
  border: 1.5px solid #999999;
}

.c3:hover {
  border-color: #666666;
}

<div
  className="c0"
>
  <input
    className="c1 c2"
    type="radio"
  />
  <span
    className="c3"
  />
</div>
`;
